hack/debug-network.sh

```diff
@@ -0,0 +1,574 @@
+#!/bin/bash
+
+# echoes args to stderr and exits
+die () {
+    echo "$*" 1>&2
+    exit 1
+}
+
+# echoes the command provided as $@ and then runs it
+echo_and_eval () {
+    echo "> $*"
+    echo ""
+    eval "$@"
+}
+
+# runs the command provided as $@, and either returns silently with
+# status 0 or else logs an error message with the command's output
+try_eval () {
+    tmpfile=`mktemp`
+    if ! eval "$@" >& $tmpfile; then
+	status=1
+	echo "ERROR: Could not run '$*':"
+	sed -e 's/^/  /' $tmpfile
+	echo ""
+    else
+	status=0
+    fi
+    rm -f $tmpfile
+    return $status
+}
+
+# The environment may contain sensitive information like passwords or private keys
+filter_env () {
+    awk '/ env:$/ { indent = index($0, "e"); skipping = 1; next } !skipping { print; } skipping { ch = substr($0, indent, 1); if (ch != " " && ch != "-") { skipping = 0; print } }'
```

Comment:
Prefer to put complicated `awk` expressions into a first-class `.awk` source file and use that instead.